util/misc.py

```diff
@@ -353,3 +354,109 @@ def get_intervals(self, c=None):
                 right = right - 1
                 features = list(self.get_features(c, left, right))
                 yield (c, left, right, len(features), features)
+
+
+def available_cpu_count():
```

Comment:
I don't think we support most of these platforms, plus this requires installing psutil. The snakemake version should be okay and is a bit conservative about different ways cpu counts are revealed to the process under different VM schemes: https://bitbucket.org/snakemake/snakemake/src/dbfd288e71e494f8964e4f1302f2e66f3b6b1eb6/snakemake/utils.py?at=master&fileviewer=file-view-default#utils.py-265